### PR TITLE
GH-87358: Add clarification about nargs and default argparse behaviour

### PR DESCRIPTION
**Pull request by savannahostrowski**
After reading through the issue's comments, it seems like all that's needed here is additional clarification in the argparse docs about how `nargs` and `default` interact.

* Issue: gh-87358

----
📚 Documentation preview 📚: https://cpython-previews--124094.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/argparse.rst
Comment (serhiy-storchaka) [inline Doc/library/argparse.rst]: This looks misleading, because it is not clear that this only relates to positional arguments.

On other hand, default is also has no effect for required optional arguments.

We could say that the default value is ignored for required arguments (there may be a reference to the `required` attribute).

**Comment by gpshead** [comment]
For future reference: Avoid rebase and force pushing as it causes codeowners assignment for all of the other changes rebased in. Just merge main in instead.

We all do this from time to time. (Even at work!) It's a GitHub git internals UX flaw. 😅

**Comment by savannahostrowski** [comment]
Yeah, oof, that's my bad! Apologies for the noise, folks 😅.

**Review by serhiy-storchaka — APPROVED** [review]
I am not sure about wording. LGTM, but it would be better if some of native speakers check it.

File: Doc/library/argparse.rst
Comment (hugovk) [inline Doc/library/argparse.rst:1126]: Aside: we have a (somewhat unusual) convention of using italics for arguments instead of literal code formatting:

https://devguide.python.org/documentation/markup/

But I see we're using literals elsewhere in this file so it's best to be consistent.

**Review by hugovk — APPROVED** [review]
Wording is good, thanks!

**Pull request merged**
